utils.py

```diff
@@ -30,6 +30,36 @@
 logger.setLevel(logging.DEBUG)
 
 
+class DenseTensor(object):
```

Comment:
Let's move these into another file, e.g format.py. 

Also, the name can be DenseFormat/DenseTensorFormat, DenseTensor sounds like a misnomer. 
Similarly for Sparse.

Reply:
done